src/Parser/Druid/Balance/Modules/Items/PromiseOfElune.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ItemDamageDone from 'Main/ItemDamageDone';
+
+const DAMAGE_MODIFIER = 0.08;
+
+class PromiseOfElune extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  solarWrathDamage = 0;
+  lunarStrikeDamage = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFeet(ITEMS.PROMISE_OF_ELUNE.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.ability.guid === SPELLS.SOLAR_WRATH_MOONKIN.id) {
+      this.solarWrathDamage += this.damageContribution(event, DAMAGE_MODIFIER);
+    }
+    if (event.ability.guid === SPELLS.LUNAR_STRIKE.id) {
+      this.lunarStrikeDamage += this.damageContribution(event, DAMAGE_MODIFIER);
+    }     
+  }
+
+  damageContribution(event, mod) {
```

Comment:
This function is already provided in core @ https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Core/calculateEffectiveDamage.js

No need to reimplement here.